components/email_aliases/resources/email_aliases_panel.tsx

```diff
@@ -0,0 +1,110 @@
+
+export const EmailAliasesPanelConnected = ({
+  emailAliasesService,
+  emailAliasesPanelHandler,
+  bindObserver,
+}: {
+  emailAliasesService: EmailAliasesServiceInterface
+  emailAliasesPanelHandler: EmailAliasesPanelHandlerInterface
+  bindObserver: (observer: EmailAliasesServiceObserverInterface) => () => void
+}) => {
+  const [authState, setAuthState] = React.useState<AuthState>({
+    status: AuthenticationStatus.kStartup,
+    email: '',
+    errorMessage: undefined,
+  })
+  const [aliasesState, setAliasesState] = React.useState<Alias[]>([])
+  React.useEffect(() => {
+    let status: AuthenticationStatus = AuthenticationStatus.kStartup
+    const observer: EmailAliasesServiceObserverInterface = {
+      onAliasesUpdated: (aliases: Alias[]) => {
+        if (status !== AuthenticationStatus.kAuthenticated) {
+          return
+        }
+        setAliasesState(aliases)
+      },
+      onAuthStateChanged: (state: AuthState) => {
+        status = state.status
+        setAuthState(state)
+        if (status !== AuthenticationStatus.kAuthenticated) {
+          setAliasesState([])
+        }
+      },
+    }
+    return bindObserver(observer)
+  }, [])
+  return (
+    <EmailAliasModal
+      onReturnToMain={(action: EmailAliasModalAction) => {
+        switch (action.type) {
+          case EmailAliasModalActionType.Cancel:
+            emailAliasesPanelHandler.onCancel()
+            break
+          case EmailAliasModalActionType.Manage:
+            emailAliasesPanelHandler.onManage()
+            break
+          case EmailAliasModalActionType.Complete:
+            emailAliasesPanelHandler.onComplete(action.email)
+            break
+        }
+      }}
+      editing={false}
+      mainEmail={authState.email}
+      aliasCount={aliasesState.length}
+      emailAliasesService={emailAliasesService}
+      bubble={true}
+    />
+  )
+}
+
+const mount = () => {
+  const rootElement = document
+    .getElementById('mountPoint')!
+    .attachShadow({ mode: 'open' })
+  const emailAliasesService = EmailAliasesService.getRemote()
+  const emailAliasesPanelHandler = EmailAliasesPanelHandler.getRemote()
+  const bindObserver = (observer: EmailAliasesServiceObserverInterface) => {
+    const observerReceiver = new EmailAliasesServiceObserverReceiver(observer)
+    const observerRemote = observerReceiver.$.bindNewPipeAndPassRemote()
+    emailAliasesService.addObserver(observerRemote)
+    return () => observerReceiver.$.close()
+  }
+  createRoot(rootElement).render(
+    <StyleSheetManager target={rootElement.getRootNode() as ShadowRoot}>
+      <BraveCoreThemeProvider>
+        <EmailAliasesPanelConnected
+          emailAliasesService={emailAliasesService}
+          emailAliasesPanelHandler={emailAliasesPanelHandler}
+          bindObserver={bindObserver}
+        />
+      </BraveCoreThemeProvider>
+    </StyleSheetManager>,
+  )
+}
+
+document.addEventListener('DOMContentLoaded', mount)
```

Comment:
tbh, I'm not sure if we need this - might as well call `mount` directly

Reply:
(you might need to move the `<script>` tag after the `<div id="mountPoint">`)

Reply:
If I call mount directly it fails because Mojo is undefined

Reply:
Hmmm that's weird

Reply:
Is this being loaded with `<script src="..." type="module">`?